### ISSUE TITLE
Accidental drop problem on c-shape argument improperly allowed

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap#2278]
If I accidently drop `list `at the wrong place

![image](https://user-images.githubusercontent.com/2138696/49436663-940dbb00-f7b1-11e8-8ac5-bfc39e86c587.png)

I get

![image](https://user-images.githubusercontent.com/2138696/49436685-a38d0400-f7b1-11e8-987f-ec1f23c0b0d7.png)

which doesn't make sense. And I can't drop `list `on a builtin block like `if`.

**Comment by brianharvey** [comment]
It _does_ make sense to drop a _variable_ (or a list item) there, if the value of the variable is a script.  But I agree that in this context dropping a list doesn't make sense.

In Brian's dream Snap<i>!</i>, a window would pop up saying "Did you mean to drop that list on the list input slot of for each?"  Super specific error messages.

**Comment by ToonTalk** [comment]
But in any case `if` shouldn't behave different from user-defined commands with c-shape inputs.

If it's OK to drop a script on `for each` then it should be for `if`` too. But it's OK to drop a script there only because `for each` uses `run`.

**Comment by brianharvey** [comment]
Yes, it's true that the builtins behave differently.  The primitive C-shaped blocks are special-cased, dating back to when Scratch had them as the only special forms in the language.

But, making it so you can drop variables onto IF isn't very high on my priority list.  Someday when things slow down.™
